### DISCUSSION TITLE
Bosch Smart Thermostat II

### DISCUSSION THREAD

**Discussion by aheartseeker** [Koenkk/zigbee2mqtt#18272] [Devices]
Hello and thank you for all the great help you are providing!

I have the Bosch BTH-RM thermostat, but the RBSH-RTH0-BAT-ZB-EU version (powered with 4 AAA batteries)
I was able to pair it easily but am encountering some issues with it and I am not able to automate a switch to control my heating.
I'm using a multifunctional relay connected to my heater and am trying to create two automations to turn a switch on and off according to the thermostat state. 
The generic thermostat does not seem to be able to switch to off state after set temperature has been reached, so I tried using the running state to achieve this. 
When running state becomes "heat" or "idle" I just trigger my automations and that's it,
I managed to do this with ZHA, but am unable to get the idle state to register in the thermostat entity with zigbee2mqtt (the device does report the running state correctly under STATE properties in zigbee2mqtt). Once activated, the thermostat reports Heating even if the running state is idle.
I've also tried better_themostat but it behaves the same and they impose that I use an external temp sensor, which is redundant for me since my thermostat is already portable.

2 other small issues:
- Humidity is N/A% (ZHA did report it)
- my firmware is 2.2.2, it would be awesome to receive a firmware update

Thank you!

**Comment by hornsbeef** [comment]
Hi!
I'm about to buy the Bosch RTH-RM230Z (the 230V variant). 
Have you had any luck getting the device to work? 

Thank you!

**Comment by aheartseeker** [comment]
Hello! 
At this time I have returned to zha with skyconnect, as zigbee2mqtt and the Sonoff Dongle-P 3.0 with the latest firmware is not yet functioning properly with my version of the thermostat. 
With my version, the thermostat entity is just not functioning properly and I had to create two automations that control the switch in accordance with hvac_action state (idle of heating).
I did manage to get the same result with z2mqtt and the sonoff device witth its factory firmware, but somehow things didn't seem to want to work again after updating the sonoff firmware and reconfiguring the integration (all other devides worked, but the thermostat would just get paired and not show any entities or anything usefull).
I  do see that the 230V version seems to have better support in zigbee2mqtt, including firmware updates, which is not the case with mine. 
I will be keeping zigbee2mqtt and the sonoff active, just in case things get better and I can switch back.
